--- a/micropython/ports/esp32/esp32_rmt.c
+++ b/micropython/ports/esp32/esp32_rmt.c
@@ -100,7 +100,7 @@ STATIC mp_obj_t esp32_rmt_make_new(const mp_obj_type_t *type, size_t n_args, s
     esp32_rmt_obj_t *self = mp_obj_malloc(esp32_rmt_obj_t, type);
 
     size_t nargs = 0;
-    uint32_t resolution_hz;
+    uint32_t resolution_hz = 1000000; // Default 1 MHz resolution
     rmt_clock_source_t clock_source = RMT_CLK_SRC_DEFAULT;
     rmt_mode_t mode = RMT_MODE_TX;
 
